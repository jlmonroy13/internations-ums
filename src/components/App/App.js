import React from 'react';
import PropTypes from 'prop-types';
import Alert from 'react-s-alert';
import { Spinner } from 'react-redux-spinner';
import 'react-s-alert/dist/s-alert-default.css';
import 'react-s-alert/dist/s-alert-css-effects/jelly.css';

class App extends React.Component {
  componentWillMount() {
    this.props.getUserInfo();
  }
  render() {
    return (
      <div>
        {this.props.children}
        <Alert stack={{ limit: 3 }} />
        <Spinner />
      </div>
    );
  }
}

App.propTypes = {
  children: PropTypes.element.isRequired,
  getUserInfo: PropTypes.func.isRequired,
};

export default App;
