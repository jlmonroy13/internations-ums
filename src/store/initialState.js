export default {
  authorization: {
    uid: '',
    isLoading: true,
  },
  events: {
    editMode: false,
    isMenuOpen: false,
    isNavOpen: false,
  },
};
