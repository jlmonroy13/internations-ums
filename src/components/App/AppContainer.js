import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { getUserInfo } from '../../actions/account';
import App from './App';

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ getUserInfo }, dispatch);
}

export default connect(null, mapDispatchToProps)(App);
