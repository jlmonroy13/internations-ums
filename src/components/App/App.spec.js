import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import { Provider } from 'react-redux';
import initialState from '../../store/initialState';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { getFirebase } from 'react-redux-firebase';

describe('<App />', () => {
  it('should render the app component', () => {
    const middlewares = [thunk.withExtraArgument(getFirebase)];
    const mockStore = configureStore(middlewares);
    const store = mockStore(initialState);
    const getUserInfo = jest.fn();
    const wrapper = renderer.create(
      <Provider store={store}>
        <App getUserInfo={getUserInfo}>
          <span />
        </App>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
