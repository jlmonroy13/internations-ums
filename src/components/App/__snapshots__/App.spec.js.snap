// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render the app component 1`] = `
<div>
  <span />
  <div
    className="s-alert-wrapper"
  />
</div>
`;
